sdk/cosmosdb/cosmos/src/common/helper.ts

```diff
@@ -255,14 +246,9 @@ export function validateResourceId(resourceId: string): boolean {
     throw new Error("Resource Id must be a string and cannot be undefined, null or empty");
   }
 
-  // if resourceId starts or ends with space throw an error
-  if (resourceId[resourceId.length - 1] === " ") {
-    throw new Error("Resource Id cannot end with space");
-  }
-
   // if resource id contains illegal characters throw an error
   if (illegalResourceIdCharacters.test(resourceId)) {
-    throw new Error("Illegal characters ['/', '\\', '?', '#'] cannot be used in resourceId");
+    throw new Error("Illegal characters ['/', '\\'] cannot be used in resourceId");
```

Comment:
Only once we have made the changes across all SDKs. Even then I would probably not lift the restriction in documentation - because it will only work end-to-end when all SDKs have fixes etc. Better to not encourage usage of these characters - but there is no need to disallow them either